google/cloud/bigquery/job/base.py

```diff
@@ -814,28 +815,36 @@ def reload(
                 ``client`` stored on the current dataset.
 
             retry (Optional[google.api_core.retry.Retry]): How to retry the RPC.
-            timeout (Optional[float]):
+            timeout (Optinal[Union[float, \
+                google.api_core.future.polling.PollingFuture._DEFAULT_VALUE, \
+            ]]):
                 The number of seconds to wait for the underlying HTTP transport
                 before using ``retry``.
         """
         client = self._require_client(client)
 
-        extra_params = {}
-        if self.location:
-            extra_params["location"] = self.location
-        span_attributes = {"path": self.path}
+        kwargs: Dict[str, Any] = {}
+        if type(timeout) is object:
```

Comment:
Let's add some comments explaining why we need to omit the timeout if we get `_DEFAULT_VALUE`. (Because we want to use the default API-level timeouts but wait indefinitely for the job to finish)

Reply:
I added explanation why `kwargs` is used here, but I feel like `Queryjob.result()` is a better place to explain the default timeout behavior, so I added it in docstring there instead.